integration/local-gadget/integration_test.go

```diff
@@ -0,0 +1,93 @@
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"flag"
+	"fmt"
+	"math/rand"
+	"os"
+	"strings"
+	"testing"
+	"time"
+
+	containercollection "github.com/kinvolk/inspektor-gadget/pkg/container-collection"
+)
+
+const (
+	ContainerRuntimeDocker     = "docker"
+	ContainerRuntimeContainerd = "containerd"
+	ContainerRuntimeCRIO       = "cri-o"
+)
+
+var supportedContainerRuntimes = []string{ContainerRuntimeDocker, ContainerRuntimeContainerd, ContainerRuntimeCRIO}
+
+var (
+	integration      = flag.Bool("integration", false, "run integration tests")
+	containerRuntime = flag.String("container-runtime", "", "allows to skip tests that are not supported on a given container-runtime")
+)
+
+func testMain(m *testing.M) int {
+	flag.Parse()
+	if !*integration {
+		fmt.Println("Skipping integration test.")
+		return 0
+	}
+
+	if *containerRuntime != "" {
+		found := false
+		for _, val := range supportedContainerRuntimes {
+			if *containerRuntime == val {
+				found = true
+				break
+			}
+		}
+
+		if !found {
+			fmt.Fprintf(os.Stderr, "Error: invalid argument '-container-runtime': %q. Valid values: %s\n",
+				*containerRuntime, strings.Join(supportedContainerRuntimes, ", "))
+			return -1
+		}
+	}
+
+	seed := time.Now().UTC().UnixNano()
+	rand.Seed(seed)
+	fmt.Printf("using random seed: %d\n", seed)
+
+	fmt.Println("Start running tests:")
+	return m.Run()
+}
+
+func TestMain(m *testing.M) {
+	os.Exit(testMain(m))
+}
+
+func TestListContainers(t *testing.T) {
+	ctx := context.Background()
+	prefix := "list-containers"
+	podID, err := runTestContainer(ctx, *containerRuntime, prefix, "docker.io/library/alpine", "sleep inf")
+	if err != nil {
+		t.Fatalf("failed to run container: %v", err)
+	}
+	defer removeTestContainer(ctx, *containerRuntime, podID)
+
+	out, err := run("local-gadget", "list-containers", "-o", "json", "--runtimes", *containerRuntime)
+	if err != nil {
+		t.Fatalf("failed to run commamd: %v", err)
+	}
+
+	var containers []containercollection.Container
+	if err = json.Unmarshal(out, &containers); err != nil {
+		t.Fatalf("failed to unmarshal: %v", err)
+	}
+
+	var found bool
+	for _, container := range containers {
+		if strings.Contains(container.Name, prefix) {
+			found = true
+		}
+	}
```

Comment:
For instance, once https://github.com/kinvolk/inspektor-gadget/pull/973 is merged, we can verify here that containers created with `kubectl-gadget` were enriched with the correct values: namespace, pod and poduid. Such verification should be omitted if containers were created via `docker`, `ctr` or ... I don't know how to create standalone CRI-O containers 😅 .

Reply:
Nice idea. As of now I don't think it would be possible to create non CRI complaint container uing CRI-O. I used runtime service or `crictl` to create such containers and that obviously have the concept of pod/sandbox. It would be so weird if it does support something like that! :smile:

Reply:
Yeah, I also think it is not possible. Maybe it is not even possible by definition 😅.